### PR TITLE
Various cleanups in src/_pytest/python.py

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
those are random clean-ups i did while tinkering in the direction of function definition

### CODE REVIEW

**Comment by nicoddemus** [comment]
Looking good so far, waiting until this is no longer WIP to give it a final review. 👍

File: src/_pytest/python.py [hunk 1]
```diff
@@ -1086,6 +1058,12 @@ def _resolve_arg_value_types(self, argnames, indirect):
                         pytrace=False,
                     )
                 valtypes[arg] = "params"
+        else:
+            raise TypeError(
+                "expected Sequence or boolean for indirect, got {type!r}".format(
```
Comment (nicoddemus) [inline src/_pytest/python.py]: Oops actually this is incorrect.

Comment (nicoddemus) [reply src/_pytest/python.py]: After fixing it (`.format(type=type(indirect))`), we still get a non-user-friendly traceback though:

```
============================================= ERRORS ==============================================
________________________________ ERROR collecting .tmp/test-sw.py _________________________________
.env36\lib\site-packages\pluggy\hooks.py:289: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
.env36\lib\site-packages\pluggy\manager.py:87: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.env36\lib\site-packages\pluggy\manager.py:81: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
src\_pytest\python.py:227: in pytest_pycollect_makeitem
    res = list(collector._genfunctions(name, obj))
src\_pytest\python.py:396: in _genfunctions
    self.ihook.pytest_generate_tests.call_extra(methods, dict(metafunc=metafunc))
.env36\lib\site-packages\pluggy\hooks.py:327: in call_extra
    return self(**kwargs)
.env36\lib\site-packages\pluggy\hooks.py:289: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
.env36\lib\site-packages\pluggy\manager.py:87: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
.env36\lib\site-packages\pluggy\manager.py:81: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
src\_pytest\python.py:128: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
src\_pytest\python.py:978: in parametrize
    arg_values_types = self._resolve_arg_value_types(argnames, indirect)
src\_pytest\python.py:1064: in _resolve_arg_value_types
    type=type(indirect)
E   TypeError: expected Sequence or boolean for indirect, got <class 'set'>
```

How about just calling `pytest.fail(..., pytrace=False)` here instead?

```
============================================= ERRORS ==============================================
________________________________ ERROR collecting .tmp/test-sw.py _________________________________
expected Sequence or boolean for indirect, got <class 'set'>
```

Not very informative, but at least we don't show a traceback which is not useful to the user at all.

We probably could at least mention "parametrize" in the argument message?

Comment (nicoddemus) [reply src/_pytest/python.py]: I noticed this after looking at the coverage; we should also add a test to ensure that line works

Comment (RonnyPfannschmidt) [reply src/_pytest/python.py]: implemented a better message and using pytest.fail now

**Pull request merged**